src/components/network-modification-tree-pane.js

```diff
@@ -210,18 +218,51 @@ export const NetworkModificationTreePane = ({
         [studyUuid, enqueueSnackbar, intlRef]
     );
 
+    const handleCopyNode = useCallback(
+        (referenceNodeId) => {
+            setSelectedNodeIdForCopy(referenceNodeId);
+            setCopyType(CopyType.COPY);
+        },
+        [setSelectedNodeIdForCopy, setCopyType]
+    );
+
+    const handleCutNode = useCallback(
+        (referenceNodeId) => {
+            setSelectedNodeIdForCopy(referenceNodeId);
+            setCopyType(CopyType.CUT);
+        },
+        [setSelectedNodeIdForCopy, setCopyType]
+    );
+
     const handlePasteNode = useCallback(
         (referenceNodeId, insertMode) => {
-            copyTreeNode(
-                studyUuid,
-                selectedNodeIdForCopy,
-                referenceNodeId,
-                insertMode
-            ).catch((errorMessage) => {
-                snackError(errorMessage, 'NodeCreateError');
-            });
+            if (CopyType.CUT === copyType) {
+                cutTreeNode(
+                    studyUuid,
+                    selectedNodeIdForCopy,
+                    referenceNodeId,
+                    insertMode
+                )
+                    .then(() => {
+                        //After the first CUT / PASTE operation, it becomes a COPY / PASTE operation
+                        setSelectedNodeIdForCopy(null);
+                        setCopyType(null);
```

Comment:
if it's becoming a copy after the cut why does copyType and selectedNodeId are set to null ? Shouldn't this go to CopyType.COPY and the id stays ?

Reply:
Forgot to modify comment, after cut/paste, we can't paste anymore